### ISSUE TITLE
"mvn clean install" failed!!!

### ISSUE THREAD

**Issue by AmramXing** [kbastani/spring-cloud-microservice-example#17]
When I run "mvn clean install", it report error.
=====================================
...
[ERROR] Failed to execute goal com.spotify:docker-maven-plugin:0.2.3:build (defa
ult) on project users-microservice: Exception caught: java.util.concurrent.Execu
tionException: com.spotify.docker.client.shaded.javax.ws.rs.ProcessingException:
 org.apache.http.conn.HttpHostConnectException: Connect to localhost:2375 [local
host/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused: connect -
> [Help 1]
...
=====================================

It seem to connect localhost:2375 error,  I don't konw how to solute it.

My OS is Windows 7 and user docker windows version ,can someone tell me how to solute this error?

**Comment by NogbadTheBad** [comment]
I think you do not have Docker running. Or if you do, you have not set up the environment so that maven knows how to talk to Docker.

The instructions in https://docs.docker.com/windows/ are very good. Although there are a lot of steps.

Look for the bit about the Docker Toolbox Terminal. Run maven from inside that terminal when it's set things up for you.

Let me know if you have better luck.

/Stephen

+1 (781) 367-6461

**Comment by cleemansen** [comment]
In my first attempt building the apps with ```mvn``` also failed with that error. In order to build successfully the environment variable ```DOCKER_HOST```must be set.

So starting the ```Docker Toolbox Terminal``` fixed the problem.

**Comment by kbastani** [comment]
Solution noted by @cleemansen

**Comment by AmramXing** [comment]
I find the problem. because I use 'mvn clean install' in Windows os, not at Linux.

**Comment by lovechenrui** [comment]
I don't know how to set  environment variable DOCKER_HOST,can someone tell me. (win7; docker toolbox)

**Comment by srinivasaraogurram** [comment]
Hi All,
I have a similar error.
But above instructions suggested to use docker for windows.
I did installed and able to run the docker for docker terminal. But what is the root cause in windows.
I did check the ports if they are in use. But ports looks good. 
If some one from spring.io can answer this. I appreciate it.
![docker error with spring boot](https://cloud.githubusercontent.com/assets/2825912/19420975/6453c39c-93bc-11e6-8267-eb63be358f3d.PNG)
Thanks 
Srini
+1 925 922 4115
